velox/exec/fuzzer/PrestoQueryRunner.cpp

```diff
@@ -579,63 +611,234 @@ std::optional<std::string> PrestoQueryRunner::toSql(
+      executeVector(sql, probeInput, buildInput, resultType));
+}
+
+std::string PrestoQueryRunner::createTable(
     const std::vector<velox::RowVectorPtr>& input,
-    const velox::RowTypePtr& resultType) {
+    const std::string& tableName) {
   auto inputType = asRowType(input[0]->type());
-  if (inputType->size() == 0) {
-    // The query doesn't need to read any columns, but it needs to see a
-    // specific number of rows. Make new 'input' as single all-null BIGINT
-    // column with as many rows as original input. This way we'll be able to
-    // create a 'tmp' table will the necessary number of rows.
-    vector_size_t numInput = 0;
-    for (const auto& v : input) {
-      numInput += v->size();
-    }
-
-    auto column = BaseVector::createNullConstant(BIGINT(), numInput, pool());
-    auto rowVector = std::make_shared<RowVector>(
-        pool(),
-        ROW({"x"}, {BIGINT()}),
-        nullptr,
-        numInput,
-        std::vector<VectorPtr>{column});
-    return executeVector(sql, {rowVector}, resultType);
-  }
-
-  // Create tmp table in Presto using DWRF file format and add a single
-  // all-null row to it.
-
   std::stringstream nullValues;
   for (auto i = 0; i < inputType->size(); ++i) {
     appendComma(i, nullValues);
     nullValues << fmt::format(
         "cast(null as {})", toTypeSql(inputType->childAt(i)));
   }
 
-  execute("DROP TABLE IF EXISTS tmp");
+  execute(fmt::format("DROP TABLE IF EXISTS {}", tableName));
 
   execute(fmt::format(
-      "CREATE TABLE tmp({}) WITH (format = 'DWRF') AS SELECT {}",
+      "CREATE TABLE {}({}) WITH (format = 'DWRF') AS SELECT {}",
+      tableName,
       folly::join(", ", inputType->names()),
       nullValues.str()));
 
   // Query Presto to find out table's location on disk.
-  auto results = execute("SELECT \"$path\" FROM tmp");
+  auto results = execute(fmt::format("SELECT \"$path\" FROM {}", tableName));
 
   auto filePath = extractSingleValue<StringView>(results);
   auto tableDirectoryPath = fs::path(filePath).parent_path();
 
   // Delete the all-null row.
-  execute("DELETE FROM tmp");
+  execute(fmt::format("DELETE FROM {}", tableName));
+
+  return tableDirectoryPath;
+}
+
+std::vector<velox::RowVectorPtr> PrestoQueryRunner::executeVector(
+    const std::string& sql,
+    const std::vector<RowVectorPtr>& probeInput,
+    const std::vector<RowVectorPtr>& buildInput,
+    const velox::RowTypePtr& resultType) {
+  auto probeType = asRowType(probeInput[0]->type());
+  if (probeType->size() == 0) {
+    auto rowVector = makeNullRows(probeInput, "x", pool());
+    return executeVector(sql, {rowVector}, buildInput, resultType);
+  }
+
+  auto buildType = asRowType(buildInput[0]->type());
+  if (probeType->size() == 0) {
+    auto rowVector = makeNullRows(buildInput, "y", pool());
+    return executeVector(sql, probeInput, {rowVector}, resultType);
+  }
+
+  // Create probe/build table in Presto using DWRF file format and add a single
+  // all-null row to it.
```

Comment:
we should move this comment to `createTable` function.

Also the comment is not correct, `add a single all-null row to it` is only a step in `createTable`, it will then move the data into the tmp table.

so essentially you're refactoring the logic into a single `createTable` function and what it does is
`Create a temp table in Presto using input data in DWRF file format `